include/rapidcheck/detail/Unicode.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+namespace rc {
+namespace detail {
+
+template<typename T, typename RandomType>
```

Comment:
I'm not Unicode expert but according to Wikipedia, _"Each UCS character is abstractly represented by a code point, which is an integer between 0 and 1,114,111"_ meaning a code point should fit into `std::uint32_t`. Since it's actually just a number, it seems like the approach used by the normal integer generators should work. However, at small sizes, you'd probably want higher frequencies of valid ASCII characters, especially alphabetic ones. One way to do it would be to just randomly pick an alphabetic character or an arbitrary codepoint with the probability for an arbitrary character becoming higher as size increases.

Another option would be to "remap" Unicode space so that alphabetic chars end up in the lower codepoints (i.e. range `[0, 26)` is a - z, or something). If you do size like in `inRange` or `arbitrary<int>`, you'd get the scaling and the shrinking automatically.